Packet++/header/SomeIpLayer.h

```diff
@@ -0,0 +1,435 @@
+
+	/**
+	 * @brief Set the Method ID
+	 *
+	 * @param[in] methodID methodID to set
+	 */
+	void setMethodID(uint16_t methodID);
+
+	/**
+	 * @brief Get the Length Field of the someIP hdr
+	 *
+	 * @return uint32_t The length field of the someIP hdr
+	 */
+	uint32_t getLengthField() const;
+
+	/**
+	 * @brief Get the requestID
+	 *
+	 * @return uint32_t returned in host endian
+	 */
+	uint32_t getRequestID() const;
+
+	/**
+	 * @brief Set the Request ID
+	 *
+	 * @param[in] requestID requestID to set
+	 */
+	void setRequestID(uint32_t requestID);
+
+	/**
+	 * @brief Get the sessionID
+	 *
+	 * @return uint16_t returned in host endian
+	 */
+	uint16_t getSessionID() const;
+
+	/**
+	 * @brief Set the Session ID
+	 *
+	 * @param[in] sessionID sessionID to set
+	 */
+	void setSessionID(uint16_t sessionID);
+
+	/**
+	 * @brief Get the clientID
+	 *
+	 * @return uint16_t returned in host endian
+	 */
+	uint16_t getClientID() const;
+
+	/**
+	 * @brief Set the Client ID
+	 *
+	 * @param[in] clientID clientID to set
+	 */
+	void setClientID(uint16_t clientID);
+
+	/**
+	 * @brief Get the protocolVersion
+	 *
+	 * @return uint8_t
+	 */
+	uint8_t getProtocolVersion() const;
+
+	/**
+	 * @brief Set the Protocol Version
+	 *
+	 * @param[in] version version to set
+	 */
+	void setProtocolVersion(uint8_t version);
+
+	/**
+	 * @brief Get the interfaceVersion
+	 *
+	 * @return uint8_t
+	 */
+	uint8_t getInterfaceVersion() const;
+
+	/**
+	 * @brief Set the Interface Version
+	 *
+	 * @param[in] version version to set
+	 */
+	void setInterfaceVersion(uint8_t version);
+
+	/**
+	 * @brief Get the message type
+	 *
+	 * @return uint8_t
+	 */
+	uint8_t getMessageType() const;
+
+	/**
+	 * @brief Set the Message Type
+	 *
+	 * @param[in] type Type to set
+	 */
+	void setMessageType(MsgType type);
+
+	/**
+	 * @brief Set the Message Type
+	 *
+	 * @param[in] type Type to set
+	 */
+	void setMessageType(uint8_t type);
+
+	/**
+	 * @brief Get the returnCode
+	 *
+	 * @return uint8_t
+	 */
+	uint8_t getReturnCode() const;
+
+	/**
+	 * @brief Set the returnCode
+	 *
+	 * @param[in] returnCode ReturnCode to set
+	 */
+	void setReturnCode(uint8_t returnCode);
+
+	/**
+	 * @brief Set the length field of the someIP hdr
+	 *
+	 * @param[in] payloadLength Length of the payload
+	 */
+	void setPayloadLength(uint32_t payloadLength);
+
+	/**
+	 * @brief Get the Header Length With Payload, as is stated in the length field of the SomeIpHdr.
+	 *
+	 * @return uint32_t Length of the complete SomeIpPdu, including header and payload
+	 */
+	uint32_t getHeaderLengthWithPayload() const;
+
+	/**
+	 * @return A pointer for the layer payload, meaning the first byte after the header
+	 */
+	uint8_t *getPduPayload() const
+	{
+		return (m_PayloadPtr == nullptr) ? m_Data + sizeof(someiphdr) : m_PayloadPtr;
+	}
+
+	/**
+	 * @return The size in bytes of the payload
+	 */
+	size_t getPduPayloadSize() const
+	{
+		return getHeaderLengthWithPayload() - sizeof(someiphdr);
+	}

```

Comment:
We already have `getLayerPayloadSize()` in `pcpp::Layer` which should give the same result if we consider the payload as a separate layer